lib/blockchainmonitor.js

```diff
@@ -170,14 +174,103 @@ BlockchainMonitor.prototype._handleTxOuts = function(data) {
         walletId: walletId,
       });
       self.storage.softResetTxHistoryCache(walletId, function() {
-        self._storeAndBroadcastNotification(notification, next);
+        self.storage.updateLastUsedOn(out.address, function() {
+          self._storeAndBroadcastNotification(notification, next);
+        });
       });
     });
   }, function(err) {
     return;
   });
 };
 
+
+BlockchainMonitor.prototype._processBlockData = function(network, data, cb) {
+  var self = this;
+
+  var block = new Bitcore.Block(new Buffer(data.rawblock, 'hex'));
+  log.debug('Processing block ' + network + ' ' + block.hash);
+
+  var txs = block.toObject().transactions;
+
+  var allAddresses = {};
+  _.each(txs, function(tx) {
+    _.each(tx.outputs, function(o) {
+      if (o.script) {
+        var s = new Bitcore.Script(new Buffer(o.script, 'hex'));
+        var a = s.toAddress(network);
+        if (a) {
+          allAddresses[a] = true;
+        }
+      }
+    });
+  });
+  allAddresses = _.keys(allAddresses);
+
+  async.eachLimit(allAddresses, 10, function(address, i_cb) {
```

Comment:
IMO it would be cleaner to make the `updateLastUsedOn()` fn to accept both a single address and an array of addresses.